addons/point_of_sale/static/src/app/components/orderline/orderline.js

```diff
@@ -69,4 +48,99 @@ export class Orderline extends Component {
             ]);
         }
     }
+
+    get line() {
+        return this.props.line;
+    }
+
+    get lineContainerClasses() {
+        return {
+            selected: this.line.isSelected() && this.props.mode === "display",
+            ...this.line.getDisplayClasses(),
+            ...(this.props.class || []),
+            "border-start": this.props.mode != "receipt" && this.line.combo_parent_id,
+            "orderline-combo fst-italic ms-4": this.line.combo_parent_id,
+            "position-relative d-flex align-items-center lh-sm cursor-pointer": true, // Keep all classes here
+        };
+    }
+
+    get lineClasses() {
+        const line = this.line;
+        const props = this.props;
+        if (line.combo_parent_id) {
+            return props.mode === "receipt" ? "px-2" : "p-2";
+        } else {
+            if (props.mode === "receipt") {
+                return line.combo_line_ids.length > 0 ? "" : "py-1";
+            } else {
+                return "p-2";
+            }
+        }
+    }
+
+    get infoListClasses() {
+        const line = this.line;
+        const props = this.props;
+        if (props.mode === "receipt") {
+            return "";
+        }
+        if (line.customer_note || line.note || line.discount || line.packLotLines?.length) {
+            return "gap-2 mt-1";
+        }
+        return "";
+    }
+
+    /**
+     * To avoid to much logic in the template, we compute all values here
+     * and use them in the template.
+     */
+    get lineValues() {
```

Comment:
To me, you should either rename this method to `lineScreenValues` or something explicitely saying it's only for the display or move this into the corresponding component